pkg/ops/rawDiskGeneration.go

```diff
@@ -0,0 +1,700 @@
+	if err != nil {
+		return "", err
+	}
+
+	resetCloudInit := "01_reset.yaml"
+
+	// Calculate the size of the state partition based on the recovery image size
+	info, err := r.config.Fs.Stat(recoveryImagePath)
+	if err != nil {
+		internal.Log.Logger.Error().Err(err).Str("source", recoveryImagePath).Msg("failed to stat recovery image")
+		return "", err
+	}
+
+	size := (info.Size()*3 + 100*1024*1024) / (1024 * 1024)
+	internal.Log.Logger.Debug().Int64("size", size).Msg("calculated state partition size")
+
+	// Create a reset config
+	// This:
+	// - Adds a state partition with the calculated size
+	// - Adds a persistent partition with the rest of the disk
+	// - If the recovery mode file is present, it will run the reset command unattended
+	// - If the reset cloud init file is present, it will remove it. Magic! So we dont get any traces of the extra config for raw images
+	conf := &schema.YipConfig{
+		Name: "Expand disk layout",
+		Stages: map[string][]schema.Stage{
+			"rootfs.before": {
+				schema.Stage{
+					Name: "Add state partition",
+					Layout: schema.Layout{
+						Device: &schema.Device{
+							Label: agentConstants.RecoveryLabel,
+						},
+						Parts: []schema.Partition{
+							{
+								FSLabel:    agentConstants.StateLabel,
+								Size:       uint(size),
+								PLabel:     agentConstants.StatePartName,
+								FileSystem: agentConstants.LinuxImgFs,
+							},
+						},
+					},
+				}, schema.Stage{
+					Name: "Add persistent partition",
+					Layout: schema.Layout{
+						Device: &schema.Device{
+							Label: agentConstants.RecoveryLabel,
+						},
+						Parts: []schema.Partition{
+							{
+								FSLabel:    agentConstants.PersistentLabel,
+								Size:       0, // It will get expanded to the end of the disk
+								PLabel:     agentConstants.PersistentPartName,
+								FileSystem: agentConstants.LinuxImgFs,
+							},
+						},
+					},
+				},
+			}, "network": {
+				schema.Stage{
+					If:   `[ -f "/run/cos/recovery_mode" ]`,
+					Name: "Run auto reset",
+					Commands: []string{
+						"kairos-agent --debug reset --unattended --reboot",
+					},
+				},
+			}, "after-reset": {
+				schema.Stage{
+					If:   `[ -f "/oem/` + resetCloudInit + `" ]`,
+					Name: "Auto remove this file",
+					Commands: []string{
+						fmt.Sprintf("rm /oem/%s", resetCloudInit),
+					},
+				},
+			},
+		},
+	}
+	yipYAML, err := yaml.Marshal(conf)
+	// Save the cloud config
+	err = r.config.Fs.WriteFile(filepath.Join(tmpDirOem, resetCloudInit), yipYAML, 0o644)
+	if err != nil {
+		internal.Log.Logger.Error().Err(err).Str("target", filepath.Join(tmpDirOem, resetCloudInit)).Msg("failed to write cloud config")
+		return "", err
+	}
+
+	OemPartitionImage := v1.Image{
+		File:       filepath.Join(r.tmpDir, "oem.img"),
+		FS:         agentConstants.LinuxImgFs,
+		Label:      agentConstants.OEMLabel,
+		Size:       64,
```

Comment:
move to a constant?

Reply:
Im trying to understand where that fixed value came from in the script :D Maybe its a constant already and didnt notice, but yeah, will do

Reply:
yes, but where did the original value came from? I wonder if the default size we have for oem is indeed 64Mb xD

Reply:
That's why they call them "magic numbers" Itxaka. They are passed from wizard to wizard :D.

Reply:
ahh found it, itsd like that from the start https://github.com/kairos-io/kairos-agent/blob/main/pkg/constants/constants.go#L54

Also this shows that the efi partition should be 64 as well!

Reply:
done!